### ISSUE TITLE
Image selection issue in Android 8.0

### ISSUE THREAD

**Issue by jatin-lakhani** [coomar2841/android-multipicker-library#120]
Hi,
I've used pickImageSingle() method to pick image from galley. I got result in onImageChoosen but image is not coming. I've also checked with getting thumbnail image but it was null.

Tried in emulator as well in real device too.

Can you please fix this as I'm using this library and its very useful in many apps.

Awaiting for your positive reply, Thanks

**Comment by coomar2841** [comment]
Can you post some logs? Also, please mention which OS version and device/emulator.

**Comment by jatin-lakhani** [comment]
I've already mention version in title 8.0.
For other version works fine.

**Comment by devqmr** [comment]
I think I face the same issue, that  @jatin-lakhani face. 
next, I attach the log:

> java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/5ddfed8e-868c-4312-abd2-7d6ff7439dde.jpg"
                                                                           at java.lang.Long.parseLong(Long.java:590)
                                                                           at java.lang.Long.valueOf(Long.java:804)
                                                                           at com.kbeanie.multipicker.core.threads.FileProcessorThread.getPathAndMimeType(FileProcessorThread.java:367)
                                                                           at com.kbeanie.multipicker.core.threads.FileProcessorThread.getAbsolutePathIfAvailable(FileProcessorThread.java:343)
                                                                           at com.kbeanie.multipicker.core.threads.FileProcessorThread.processFile(FileProcessorThread.java:169)
                                                                           at com.kbeanie.multipicker.core.threads.FileProcessorThread.processFiles(FileProcessorThread.java:105)
                                                                           at com.kbeanie.multipicker.core.threads.FileProcessorThread.run(FileProcessorThread.java:79)
                                                                           at com.kbeanie.multipicker.core.threads.ImageProcessorThread.run(ImageProcessorThread.java:41)

**Comment by devqmr** [comment]
This issue happens only with the Download folder where the id of file after document return as number for android sdk before Oreo but in Oreo it return as full path string 
```
getOriginalPath > content://com.android.providers.downloads.documents/document/678
content: //com.android.providers.downloads.documents/document/raw%3A%2Fstorage%2Femulated%2F0%2FDownload%2F5ddfed8e-868c-4312-abd2-7d6ff7439dde.jpg
```

``` 
678
raw%3A%2Fstorage%2Femulated%2F0%2FDownload%2F5ddfed8e-868c-4312-abd2-7d6ff7439dde.jpg
```

**Comment by coomar2841** [comment]
@devqmr Thanks for the info. I will look into this over the weekend.

**Comment by coomar2841** [comment]
Can you let me know how did you put an image in the downloads folder? Was it through any specific application?

**Comment by zeeshanchicsol** [comment]
Facing the same issue on 8.0+  . dear can you please fix this ?

>  java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/Stylish Cute Baby.jpg"
>         at java.lang.Long.parseLong(Long.java:590)
>         at java.lang.Long.valueOf(Long.java:804)
>         at com.kbeanie.multipicker.core.threads.FileProcessorThread.getPathAndMimeType(FileProcessorThread.java:367)
>         at com.kbeanie.multipicker.core.threads.FileProcessorThread.getAbsolutePathIfAvailable(FileProcessorThread.java:343)
>         at com.kbeanie.multipicker.core.threads.FileProcessorThread.processFile(FileProcessorThread.java:169)
>         at com.kbeanie.multipicker.core.threads.FileProcessorThread.processFiles(FileProcessorThread.java:105)
>         at com.kbeanie.multipicker.core.threads.FileProcessorThread.run(FileProcessorThread.java:79)
>         at com.kbeanie.multipicker.core.threads.ImageProcessorThread.run(ImageProcessorThread.java:41)

**Comment by zeeshanchicsol** [comment]
@coomar2841   ye crash appears when selected from downloads folder. from camera folder it works fine.

**Comment by akash8511** [comment]
I also facing the same issue in Android O.
    java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/myfile.m4v"
        at java.lang.Long.parseLong(Long.java:590)
        at java.lang.Long.valueOf(Long.java:804)
        at com.kbeanie.multipicker.core.threads.FileProcessorThread.getPathAndMimeType(FileProcessorThread.java:367)
        at com.kbeanie.multipicker.core.threads.FileProcessorThread.getAbsolutePathIfAvailable(FileProcessorThread.java:343)
        at com.kbeanie.multipicker.core.threads.FileProcessorThread.processFile(FileProcessorThread.java:169)
        at com.kbeanie.multipicker.core.threads.FileProcessorThread.processFiles(FileProcessorThread.java:105)
        at com.kbeanie.multipicker.core.threads.FileProcessorThread.run(FileProcessorThread.java:79)
        at com.kbeanie.multipicker.core.threads.VideoProcessorThread.run(VideoProcessorThread.java:44)

**Comment by akash8511** [comment]
Hey,
I have checked it my device(Oneplush 5) and getting the same issue.

**Comment by h-tolou** [comment]
i have same problem and solved it.
add this line 
 if (!TextUtils.isEmpty(id))
                    if (id.startsWith("raw:")) {
                        return id.replaceFirst("raw:", "");
                    }
in my code


if (isDownloadsDocument(uri)) {

                final String id = DocumentsContract.getDocumentId(uri);
                if (!TextUtils.isEmpty(id))
                    if (id.startsWith("raw:")) {
                        return id.replaceFirst("raw:", "");
                    }
                final Uri contentUri = ContentUris.withAppendedId(
                        Uri.parse("content://downloads/public_downloads"), Long.valueOf(id));

                return getDataColumn(context, contentUri, null, null);
            }

**Comment by coomar2841** [comment]
Thanks. Can you tell me which file is this?

**Comment by vishrut2207** [comment]
I have the same problem and solved it.
I don't know it is the valid solution or not but try this.
I think it may work for you.

**Replace
getAbsolutePathIfAvailable in FileProcessorThread**
```
private ChosenFile getAbsolutePathIfAvailable(ChosenFile file) {

       String[] projection = {MediaStore.MediaColumns.DATA, MediaStore.MediaColumns.DISPLAY_NAME, MediaStore.MediaColumns.MIME_TYPE};

 // Workaround for various implementations for Google Photos/Picasa
       if (file.getQueryUri().startsWith("content://com.android.gallery3d.provider")) {
           file.setOriginalPath(Uri.parse(file.getQueryUri().replace("com.android.gallery3d", "com.google.android.gallery3d")).toString());
       } else {
           file.setOriginalPath(file.getQueryUri());
       }

       // Try to see if there's a cached local copy that is available
       if (file.getOriginalPath().startsWith("content://")) {
           try {
               Cursor cursor = context.getContentResolver().query(Uri.parse(file.getOriginalPath()), projection,
                       null, null, null);
               cursor.moveToFirst();
               try {
                   // Samsung Bug
                   if (!file.getOriginalPath().contains("com.sec.android.gallery3d.provider")) {
                       String path = cursor.getString(cursor
                               .getColumnIndexOrThrow(MediaStore.MediaColumns.DATA));
                       LogUtils.d(TAG, "processFile: Path: " + path);
                       if (path != null) {
                           file.setOriginalPath(path);
                       }
                   }
               } catch (Exception e) {
                   e.printStackTrace();
               }
               try {
                   String displayName = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.DISPLAY_NAME));
                   if (displayName != null) {
                       file.setDisplayName(displayName);
                   }
               } catch (Exception e) {
                   e.printStackTrace();
               }
               String mimeType = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.MIME_TYPE));
               if (mimeType != null) {
                   file.setMimeType(mimeType);
               }
               cursor.close();
           } catch (Exception e) {
               e.printStackTrace();
           }
       }

       if (!TextUtils.isEmpty(Uri.parse(file.getOriginalPath()).toString())) {
           if (!file.getQueryUri().contains("raw%")) {
               
               // Check if DownloadsDocument in which case, we can get the local copy by using the content provider
               if (file.getOriginalPath().startsWith("content:") && isDownloadsDocument(Uri.parse(file.getOriginalPath()))) {

                   if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
                       String[] data = getPathAndMimeType(file);
                       if (data[0] != null) {
                           file.setOriginalPath(data[0]);
                       }
                       if (data[1] != null) {
                           file.setMimeType(data[1]);
                       }
                   }
               }

           }
       }

       return file;
   }
```

**Comment by zeeshanchicsol** [comment]
Hi @coomar2841  above given(@vishrut2207 ) code works like charm . dear can you please update library ?

**Comment by coomar2841** [comment]
Will update this in a day or 2. Thanks,

**Comment by coomar2841** [comment]
Will try to update this today. Sorry, have been really busy these days.

**Comment by Tooto** [comment]
i can check in 12 hours. Test device stayed at work

**Comment by Tooto** [comment]
Looks like crash fixed! At least at this apk

**Comment by coomar2841** [comment]
Did you get the actual image?  Anyone else tried it yet?

**Comment by coomar2841** [comment]
This should be fixed in 1.6. Please re-open if issue exists.
